GDJS/Runtime/types/project-data.d.ts

```diff
@@ -90,6 +95,12 @@ declare interface LayoutData {
 declare interface EventsFunctionsExtensionData {
   name: string;
   eventsBasedObjects: EventsBasedObjectData[];
+  globalVariables: RootVariableData[];
+  sceneVariables: RootVariableData[];
+}
+
+declare interface LayoutAndExtensionsData extends LayoutData {
+  usedExtensionsWithVariablesData?: ?EventsFunctionsExtensionData[];
 }
```

Comment:
I don't like inheritance! It makes a bit fuzzy to understand things and give the false impression that you can pass a LayoutAndExtensionData where a LayoutData is expected - which would be super weird and surely a logic error.

Instead, why not:

```suggestion
declare interface LayoutAndExtensionsData {
  layoutData: LayoutData;
  usedExtensionsWithVariablesData: EventsFunctionsExtensionData[];
}
```

Reply:
ok, it will break the compatibility on `loadFromScene` but maybe it's fine.

Reply:
It's indeed a public API breaking change, but using this API means you're doing some dark magic with the scenes - I don't think it will be a problem to ask people doing so to adapt their code :)